packages/storage-driver-gcs/src/index.ts

```diff
@@ -88,6 +109,64 @@ export class DriverGCS implements Driver {
 			query = nextQuery;
 		}
 	}
+
+	get tusExtensions() {
+		return ['creation', 'termination', 'expiration'];
+	}
+
+	async createChunkedUpload(filepath: string, context: ChunkedUploadContext) {
+		const file = this.file(this.fullPath(filepath));
+
+		const [uri] = await file.createResumableUpload();
+
+		context.metadata!['uri'] = uri;
+
+		return context;
+	}
+
+	async writeChunk(filepath: string, content: Readable, offset: number, context: ChunkedUploadContext) {
+		const file = this.file(this.fullPath(filepath));
+
+		const stream = file
+			.createWriteStream({
+				chunkSize: this.preferredChunkSize,
+				uri: context.metadata!['uri'] as string,
+				offset,
+				isPartialUpload: true,
+				resumeCRC32C: context.metadata!['crc32'] as string,
+				metadata: {
+					contentLength: context.size || 0,
+					metadata: {
+						size: context.size || 0,
```

Comment:
Any specific metadata we want to be setting on the file? iirc metadata is a required field/param with at least one key/value.

Reply:
Disregard, upon further testing this does not seem to be the case any longer